CenterForOpenScience/osf.io
# [ENG-2633] Migrate egap fields

## Purpose
EGAP Registration ID is moving out of schema and into the new `provider_specific_metadata` space for a Registration to enable contributors to submit their own EGAP registrations. Similarly, the "Timestamp of original registration" schema field will no longer be relevant once users submit their own registrations directly, as the "Date Registered" field will be a viable source of truth.

## Changes
* Add a management command to backfill the EGAP Registration ID and registered_date fields for EGAP registrations based on data in the registration_responses.
* Update the EGAP registration schema to omit the questions for those values.

Note, the registered_date is being populated following the same method used in the import_EGAP management command.

Note, following precedent and simply removing the obsolete questions without bumping the subsequent question numbers. Possibly, this is to avoid breaking the `create_EGAP_json` script?

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify that the script runs correctly
- Verify that EGAP Registration ID and Date Registered are correctly copied 

## Documentation



## Side Effects



## Ticket
https://openscience.atlassian.net/browse/ENG-2633
